compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -793,6 +791,7 @@ public void visit(BLangPackage pkgNode) {
         pkgNode.initFunction = rewrite(pkgNode.initFunction, env);
         pkgNode.startFunction = rewrite(pkgNode.startFunction, env);
         pkgNode.stopFunction = rewrite(pkgNode.stopFunction, env);
+        pkgNode.functions = rewrite(pkgNode.functions, env);
```

Comment:
Note: when query-expressions are at the module level, it will get desugared through the init function. And within that process, it will create define several lambda functions and those won't get desugared (visited) if the `pkgNode.functions` get re-written before `pkgNode.initFunction`.

Reply:
Should this lambda creation done in `desugarGlobalVariables` function?

Reply:
`desugarGlobalVariables` will only add the global variable to the `init` function body, right? And, that added variable will get desugared when we re-write init body stmts, yeah?

Reply:
It does perform desugaring parts besides adding to the init function. Check this out  (configurables and listener) https://github.com/rdhananjaya/ballerina-lang/blob/c25c6e40eed05bef4bfb9d74d005943f57bec41e/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L911

Reply:
Yes. However, at this stage, it's just another `VARIABLE` right? I mean, it's not like a `QUERY_VARIABLE`. And the rhs expr makes it a query. So, is it ok to check the expr NodeKind and re-write it, only if it's a query-expr?

And even if we do, at this point the enclosing invocable of the re-written `stmt-expr`  will be null, right? (since it's not yet added to init body, and still at the package level).